### ISSUE TITLE
Endless loop of "ACTION Regenerating Makefile" during npm install on 32-bit Ubuntu

### ISSUE THREAD

**Issue by nfriedly** [theturtle32/WebSocket-Node#231]
Hi, 

@johnfelipe reported [an issue](https://github.com/watson-developer-cloud/speech-to-text-nodejs/issues/104) where installing a demo that depends on `websocket` gets into an endless loop of regenerating the makefile. It also seems to behave the same when installing the latest version as a standalone:

```
felipe@felipeurrego:~$ mkdir empty
felipe@felipeurrego:~$ cd empty/
felipe@felipeurrego:~/empty$ npm install websocket
|
> websocket@1.0.22 install /home/felipe/empty/node_modules/websocket
> (node-gyp rebuild 2> builderror.log) || (exit 0)

make: Entering directory `/home/felipe/empty/node_modules/websocket/build'
  ACTION Regenerating Makefile
make: Leaving directory `/home/felipe/empty/node_modules/websocket/build'
make: Entering directory `/home/felipe/empty/node_modules/websocket/build'
  ACTION Regenerating Makefile
make: Leaving directory `/home/felipe/empty/node_modules/websocket/build'
make: Entering directory `/home/felipe/empty/node_modules/websocket/build'
  ACTION Regenerating Makefile
make: Leaving directory `/home/felipe/empty/node_modules/websocket/build'
make: Entering directory `/home/felipe/empty/node_modules/websocket/build'
  ACTION Regenerating Makefile
make: Leaving directory `/home/felipe/empty/node_modules/websocket/build'
make: Entering directory `/home/felipe/empty/node_modules/websocket/build'
  ACTION Regenerating Makefile
make: Leaving directory `/home/felipe/empty/node_modules/websocket/build'
make: Entering directory `/home/felipe/empty/node_modules/websocket/build'
  ACTION Regenerating Makefile
make: Leaving directory `/home/felipe/empty/node_modules/websocket/build'
make: Entering directory `/home/felipe/empty/node_modules/websocket/build'
  ACTION Regenerating Makefile
make: Leaving directory `/home/felipe/empty/node_modules/websocket/build'
make: Entering directory `/home/felipe/empty/node_modules/websocket/build'
  ACTION Regenerating Makefile
make: Leaving directory `/home/felipe/empty/node_modules/websocket/build'
make: Entering directory `/home/felipe/empty/node_modules/websocket/build'
  ACTION Regenerating Makefile
make: Leaving directory `/home/felipe/empty/node_modules/websocket/build'
make: Entering directory `/home/felipe/empty/node_modules/websocket/build'
  ACTION Regenerating Makefile
make: Leaving directory `/home/felipe/empty/node_modules/websocket/build'
make: Entering directory `/home/felipe/empty/node_modules/websocket/build'
  ACTION Regenerating Makefile
make: Leaving directory `/home/felipe/empty/node_modules/websocket/build'
make: Entering directory `/home/felipe/empty/node_modules/websocket/build'
  ACTION Regenerating Makefile
```
(this continues until canceled by ctrl-c)

I'm bring up a 32-bit ubuntu 14.04 box right now to see if I can replicate this locally

**Comment by johnfelipe** [comment]
sudo apt-get update

	sudo apt-get install libc-bin libc-dev-bin libc6-dev glibc-doc

	sudo apt-get install curl git

	curl -sL https://deb.nodesource.com/setup_4.x | sudo -E bash -

	sudo apt-get install nodejs

	git clone https://github.com/watson-developer-cloud/speech-to-text-nodejs.git

	cd speech-to-text-nodejs/

i put 
  username: process.env.STT_USERNAME || '<username>',
  password: process.env.STT_PASSWORD || '<password>'

	npm install

**Comment by theturtle32** [comment]
Thanks for the report! I think this was coincidentally resolved when I released 1.0.23 earlier tonight in order to bump `nan` dependency.

```
ubuntu@ip-10-180-220-48:~/speech-to-text-nodejs-master$ lsb_release -a
No LSB modules are available.
Distributor ID: Ubuntu
Description:    Ubuntu 14.04.4 LTS
Release:    14.04
Codename:   trusty
ubuntu@ip-10-180-220-48:~/speech-to-text-nodejs-master$ node --version
v4.4.4
ubuntu@ip-10-180-220-48:~/speech-to-text-nodejs-master$ npm --version
2.15.1
ubuntu@ip-10-180-220-48:~/speech-to-text-nodejs-master$ uname -a
Linux ip-10-180-220-48 3.13.0-86-generic #130-Ubuntu SMP Mon Apr 18 18:29:07 UTC 2016 i686 i686 i686 GNU/Linux

ubuntu@ip-10-180-220-48:~/speech-to-text-nodejs-master$ npm install
npm WARN peerDependencies The peer dependency browserify@>= 2.3.0 < 14 included from browserify-shim will no
npm WARN peerDependencies longer be automatically installed to fulfill the peerDependency
npm WARN peerDependencies in npm 3+. Your application will need to depend on it explicitly.
npm WARN deprecated jade@0.26.3: Jade has been renamed to pug, please install the latest version of pug instead of jade
npm WARN optional dep failed, continuing fsevents@1.0.12
npm WARN deprecated graceful-fs@2.0.3: graceful-fs v3.0.0 and before will fail on node releases >= v7.0. Please update to graceful-fs@^4.0.0 as soon as possible. Use 'npm ls graceful-fs' to find it in the tree.

> websocket@1.0.23 install /home/ubuntu/speech-to-text-nodejs-master/node_modules/watson-developer-cloud/node_modules/websocket
> (node-gyp rebuild 2> builderror.log) || (exit 0)

vcap_services@0.1.7 node_modules/vcap_services

dotenv@2.0.0 node_modules/dotenv

express-secure-only@0.2.1 node_modules/express-secure-only

cookie-parser@1.4.1 node_modules/cookie-parser
├── cookie-signature@1.0.6
└── cookie@0.2.3

express-rate-limit@2.3.1 node_modules/express-rate-limit
└── defaults@1.0.3 (clone@1.0.2)

csurf@1.8.3 node_modules/csurf
├── cookie@0.1.3
├── cookie-signature@1.0.6
├── http-errors@1.3.1 (inherits@2.0.1, statuses@1.3.0)
└── csrf@3.0.1 (rndm@1.2.0, base64-url@1.2.1, scmp@1.0.0, uid-safe@2.1.0)

ejs@2.4.1 node_modules/ejs

express@4.13.4 node_modules/express
├── cookie-signature@1.0.6
├── content-type@1.0.2
├── escape-html@1.0.3
├── array-flatten@1.1.1
├── utils-merge@1.0.0
├── methods@1.1.2
├── merge-descriptors@1.0.1
├── vary@1.0.1
├── cookie@0.1.5
├── fresh@0.3.0
├── range-parser@1.0.3
├── parseurl@1.3.1
├── content-disposition@0.5.1
├── etag@1.7.0
├── path-to-regexp@0.1.7
├── serve-static@1.10.2
├── depd@1.1.0
├── qs@4.0.0
├── finalhandler@0.4.1 (unpipe@1.0.0)
├── on-finished@2.3.0 (ee-first@1.1.1)
├── debug@2.2.0 (ms@0.7.1)
├── proxy-addr@1.0.10 (forwarded@0.1.0, ipaddr.js@1.0.5)
├── send@0.13.1 (destroy@1.0.4, statuses@1.2.1, ms@0.7.1, mime@1.3.4, http-errors@1.3.1)
├── type-is@1.6.13 (media-typer@0.3.0, mime-types@2.1.11)
└── accepts@1.2.13 (negotiator@0.5.3, mime-types@2.1.11)

body-parser@1.14.2 node_modules/body-parser
├── content-type@1.0.2
├── bytes@2.2.0
├── depd@1.1.0
├── qs@5.2.0
├── http-errors@1.3.1 (inherits@2.0.1, statuses@1.3.0)
├── on-finished@2.3.0 (ee-first@1.1.1)
├── raw-body@2.1.6 (unpipe@1.0.0, bytes@2.3.0)
├── iconv-lite@0.4.13
├── debug@2.2.0 (ms@0.7.1)
└── type-is@1.6.13 (media-typer@0.3.0, mime-types@2.1.11)

helmet@1.3.0 node_modules/helmet
├── hpkp@1.1.0
├── nocache@1.0.0
├── dont-sniff-mimetype@1.0.0
├── hide-powered-by@1.0.0
├── ienoopen@1.0.0
├── dns-prefetch-control@0.1.0
├── x-xss-protection@1.0.0
├── frameguard@1.1.0 (lodash.isstring@4.0.1)
├── hsts@1.0.0 (core-util-is@1.0.2)
├── connect@3.4.1 (utils-merge@1.0.0, parseurl@1.3.1, debug@2.2.0, finalhandler@0.4.1)
└── helmet-csp@1.1.0 (lodash.isfunction@3.0.8, platform@1.3.1, camelize@1.0.0, lodash.assign@4.0.4, content-security-policy-builder@1.0.0, lodash.some@4.2.0, lodash.reduce@4.2.0)

supertest@1.2.0 node_modules/supertest
├── methods@1.1.2
└── superagent@1.8.3 (component-emitter@1.2.1, cookiejar@2.0.6, reduce-component@1.0.1, extend@3.0.0, mime@1.3.4, formidable@1.0.17, qs@2.3.3, debug@2.2.0, readable-stream@1.0.27-1, form-data@1.0.0-rc3)

cf-deployment-tracker-client@0.0.8 node_modules/cf-deployment-tracker-client
└── restler@3.3.0 (yaml@0.2.3, qs@1.2.0, iconv-lite@0.2.11, xml2js@0.4.0)

mocha@2.4.5 node_modules/mocha
├── escape-string-regexp@1.0.2
├── commander@2.3.0
├── diff@1.4.0
├── supports-color@1.2.0
├── growl@1.8.1
├── debug@2.2.0 (ms@0.7.1)
├── mkdirp@0.5.1 (minimist@0.0.8)
├── glob@3.2.3 (inherits@2.0.1, graceful-fs@2.0.3, minimatch@0.2.14)
└── jade@0.26.3 (commander@0.6.1, mkdirp@0.3.0)

watson-developer-cloud@1.8.0 node_modules/watson-developer-cloud
├── cookie@0.2.3
├── isstream@0.1.2
├── extend@3.0.0
├── string@3.3.1
├── object.omit@2.0.0 (is-extendable@0.1.1, for-own@0.1.4)
├── object.pick@1.1.2 (isobject@2.1.0)
├── csv-stringify@1.0.4 (lodash.get@4.3.0)
├── websocket@1.0.23 (yaeti@0.0.4, typedarray-to-buffer@3.1.2, debug@2.2.0, nan@2.3.3)
├── request@2.72.0 (aws-sign2@0.6.0, tunnel-agent@0.4.3, oauth-sign@0.8.2, forever-agent@0.6.1, is-typedarray@1.0.0, caseless@0.11.0, stringstream@0.0.5, aws4@1.4.1, json-stringify-safe@5.0.1, tough-cookie@2.2.2, node-uuid@1.4.7, qs@6.1.0, mime-types@2.1.11, combined-stream@1.0.5, form-data@1.0.0-rc4, bl@1.1.2, hawk@3.1.3, http-signature@1.1.1, har-validator@2.0.6)
└── solr-client@0.6.0 (duplexer@0.1.1, httperror@0.2.3, JSONStream@1.0.7, request@2.63.0, json-bigint@0.1.4)

nock@7.7.3 node_modules/nock
├── propagate@0.3.1
├── json-stringify-safe@5.0.1
├── deep-equal@1.0.1
├── qs@6.2.0
├── debug@2.2.0 (ms@0.7.1)
├── mkdirp@0.5.1 (minimist@0.0.8)
├── chai@3.5.0 (assertion-error@1.0.1, type-detect@1.0.0, deep-eql@0.1.3)
└── lodash@3.10.1

express-browserify@1.0.2 node_modules/express-browserify
├── typed-args@1.0.0
├── lodash@3.10.1
├── browserify@11.2.0 (string_decoder@0.10.31, isarray@0.0.1, https-browserify@0.0.1, tty-browserify@0.0.0, constants-browserify@0.0.1, path-browserify@0.0.0, inherits@2.0.1, punycode@1.4.1, builtins@0.0.7, browser-resolve@1.11.1, os-browserify@0.1.2, htmlescape@1.1.1, process@0.11.3, commondir@0.0.1, defined@1.0.0, xtend@4.0.1, stream-browserify@2.0.1, shell-quote@0.0.1, assert@1.3.0, domain-browser@1.1.7, querystring-es3@0.2.1, timers-browserify@1.4.2, deps-sort@1.3.9, util@0.10.3, events@1.0.2, parents@1.0.1, vm-browserify@0.0.4, has@1.0.1, console-browserify@1.1.0, url@0.10.3, subarg@1.0.0, through2@1.1.1, concat-stream@1.4.10, duplexer2@0.0.2, shasum@1.0.2, read-only-stream@1.1.1, buffer@3.6.0, glob@4.5.3, stream-http@1.7.1, JSONStream@1.1.1, resolve@1.1.7, syntax-error@1.1.6, browserify-zlib@0.1.4, labeled-stream-splicer@1.0.2, browser-pack@5.0.1, insert-module-globals@6.6.3, crypto-browserify@3.11.0, readable-stream@2.1.3, module-deps@3.9.1)
└── watchify@3.7.0 (defined@1.0.0, xtend@4.0.1, through2@2.0.1, outpipe@1.1.1, anymatch@1.3.0, chokidar@1.5.0, browserify@13.0.1)

browserify@13.0.1 node_modules/browserify
├── https-browserify@0.0.1
├── tty-browserify@0.0.0
├── path-browserify@0.0.0
├── duplexer2@0.1.4
├── punycode@1.4.1
├── browser-resolve@1.11.1
├── constants-browserify@1.0.0
├── string_decoder@0.10.31
├── os-browserify@0.1.2
├── inherits@2.0.1
├── htmlescape@1.1.1
├── process@0.11.3
├── stream-browserify@2.0.1
├── assert@1.3.0
├── defined@1.0.0
├── domain-browser@1.1.7
├── xtend@4.0.1
├── read-only-stream@2.0.0
├── querystring-es3@0.2.1
├── timers-browserify@1.4.2
├── util@0.10.3
├── deps-sort@2.0.0
├── parents@1.0.1 (path-platform@0.11.15)
├── events@1.1.0
├── vm-browserify@0.0.4 (indexof@0.0.1)
├── has@1.0.1 (function-bind@1.1.0)
├── console-browserify@1.1.0 (date-now@0.1.4)
├── shell-quote@1.6.0 (array-filter@0.0.1, jsonify@0.0.0, array-map@0.0.0, array-reduce@0.0.0)
├── url@0.11.0 (punycode@1.3.2, querystring@0.2.0)
├── subarg@1.0.0 (minimist@1.2.0)
├── labeled-stream-splicer@2.0.0 (isarray@0.0.1, stream-splicer@2.0.0)
├── shasum@1.0.2 (sha.js@2.4.5, json-stable-stringify@0.0.1)
├── stream-http@2.3.0 (builtin-status-codes@2.0.0, to-arraybuffer@1.0.1)
├── through2@2.0.1 (readable-stream@2.0.6)
├── concat-stream@1.5.1 (typedarray@0.0.6, readable-stream@2.0.6)
├── glob@5.0.15 (path-is-absolute@1.0.0, once@1.3.3, inflight@1.0.5, minimatch@3.0.0)
├── JSONStream@1.1.1 (through@2.3.8, jsonparse@1.2.0)
├── buffer@4.6.0 (ieee754@1.1.6, isarray@1.0.0, base64-js@1.1.2)
├── browserify-zlib@0.1.4 (pako@0.2.8)
├── syntax-error@1.1.6 (acorn@2.7.0)
├── resolve@1.1.7
├── browser-pack@6.0.1 (umd@3.0.1, combine-source-map@0.7.2)
├── insert-module-globals@7.0.1 (is-buffer@1.1.3, combine-source-map@0.7.2, lexical-scope@1.2.0)
├── crypto-browserify@3.11.0 (create-hmac@1.1.4, randombytes@2.0.3, pbkdf2@3.0.4, create-hash@1.1.2, diffie-hellman@5.0.2, create-ecdh@4.0.0, browserify-cipher@1.0.0, browserify-sign@4.0.0, public-encrypt@4.0.0)
├── readable-stream@2.1.3 (buffer-shims@1.0.0, process-nextick-args@1.0.7, util-deprecate@1.0.2, core-util-is@1.0.2, isarray@1.0.0)
└── module-deps@4.0.5 (stream-combiner2@1.1.1, detective@4.3.1)

browserify-shim@3.8.12 node_modules/browserify-shim
├── through@2.3.8
├── mothership@0.2.0 (find-parent-dir@0.3.0)
├── resolve@0.6.3
├── rename-function-calls@0.1.1 (detective@3.1.0)
└── exposify@0.4.3 (map-obj@1.0.1, has-require@1.1.0, globo@1.0.2, transformify@0.1.2, through2@0.4.2, replace-requires@1.0.3)

eslint@2.10.2 node_modules/eslint
├── path-is-absolute@1.0.0
├── imurmurhash@0.1.4
├── estraverse@4.2.0
├── pluralize@1.2.1
├── ignore@3.1.2
├── path-is-inside@1.0.1
├── globals@9.6.0
├── esutils@2.0.2
├── strip-json-comments@1.0.4
├── progress@1.1.8
├── text-table@0.2.0
├── user-home@2.0.0 (os-homedir@1.0.1)
├── is-resolvable@1.0.0 (tryit@1.0.2)
├── debug@2.2.0 (ms@0.7.1)
├── optionator@0.8.1 (fast-levenshtein@1.1.3, type-check@0.3.2, levn@0.3.0, wordwrap@1.0.0, deep-is@0.1.3, prelude-ls@1.1.2)
├── chalk@1.1.3 (escape-string-regexp@1.0.5, ansi-styles@2.2.1, supports-color@2.0.0, has-ansi@2.0.0, strip-ansi@3.0.1)
├── doctrine@1.2.1 (isarray@1.0.0, esutils@1.1.6)
├── json-stable-stringify@1.0.1 (jsonify@0.0.0)
├── require-uncached@1.0.2 (resolve-from@1.0.1, caller-path@0.1.0)
├── mkdirp@0.5.1 (minimist@0.0.8)
├── shelljs@0.6.0
├── concat-stream@1.5.1 (inherits@2.0.1, typedarray@0.0.6, readable-stream@2.0.6)
├── glob@7.0.3 (inherits@2.0.1, once@1.3.3, inflight@1.0.5, minimatch@3.0.0)
├── espree@3.1.4 (acorn-jsx@3.0.1, acorn@3.1.0)
├── is-my-json-valid@2.13.1 (jsonpointer@2.0.0, generate-function@2.0.0, xtend@4.0.1, generate-object-property@1.2.0)
├── inquirer@0.12.0 (ansi-regex@2.0.0, strip-ansi@3.0.1, ansi-escapes@1.4.0, through@2.3.8, rx-lite@3.1.2, cli-width@2.1.0, figures@1.7.0, run-async@0.1.0, readline2@1.0.1, string-width@1.0.1, cli-cursor@1.0.2)
├── js-yaml@3.6.1 (esprima@2.7.2, argparse@1.0.7)
├── table@3.7.8 (slice-ansi@0.0.4, tv4@1.2.7, xregexp@3.1.1, strip-ansi@3.0.1, string-width@1.0.1, bluebird@3.4.0)
├── file-entry-cache@1.2.4 (object-assign@4.1.0, flat-cache@1.0.10)
├── es6-map@0.1.3 (d@0.1.1, es6-symbol@3.0.2, event-emitter@0.3.4, es6-iterator@2.0.0, es6-set@0.1.4, es5-ext@0.10.11)
├── escope@3.6.0 (esrecurse@4.1.0, es6-weak-map@2.0.1)
└── lodash@4.12.0
```

**Comment by theturtle32** [comment]
Just tried it again using the packaged Node 4.4.4 from the NodeSource apt repo, as @johnfelipe did, and it still works fine.
